molgenis-api-data/src/main/java/org/molgenis/api/data/v3/DataServiceV3Impl.java

```diff
@@ -72,6 +77,53 @@ public Entity find(String entityTypeId, String entityId, Selection filter, Selec
     return entity;
   }
 
+  @Transactional(readOnly = true)
+  @Override
+  public Entities findSubresources(
+      String entityTypeId,
+      String entityId,
+      String attributeName,
+      @Nullable @CheckForNull Query query,
+      Selection filter,
+      Selection expand,
+      Sort sort,
+      int size,
+      int number) {
+    Repository<Entity> repository = getRepository(entityTypeId);
+    EntityType entityType = repository.getEntityType();
+    Object typedEntityId = toTypedEntityId(entityType, entityId);
+
+    // Check if the attribute is suitable for this endpoint
+    Attribute refAttribute = entityType.getAttribute(attributeName);
+    if (refAttribute == null) {
+      throw new UnknownAttributeException(entityType, attributeName);
+    }
+    if (!EntityTypeUtils.isMultipleReferenceType(refAttribute)) {
+      throw new UnsupportedAttributeTypeException(refAttribute);
+    }
+
+    // get the mref ID's for this entity
+    EntityType refEntityType = refAttribute.getRefEntity();
+    Fetch refFetch = new Fetch().field(refEntityType.getIdAttribute().getName());
+    Fetch fetch = new Fetch().field(refAttribute.getName(), refFetch);
+    Entity entity = repository.findOneById(typedEntityId, fetch);
+    List<Object> refEntityIds =
+        StreamSupport.stream(entity.getEntities(attributeName).spliterator(), false)
+            .map(ref -> ref.getIdValue())
```

Comment:
Please add a unit test for this case

Reply:
Request fails with internal server error if refEntityIds is empty (see other comment for example)